src/data/domain/PersonBountyEventDataRepository.java

```diff
@@ -0,0 +1,111 @@
+package data.domain;
+
+import com.fs.starfarer.api.impl.campaign.shared.PersonBountyEventData;
+import com.fs.starfarer.api.impl.campaign.shared.SharedData;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+
+/**
+ * Repository for streamlining access to {@link SharedData#getPersonBountyEventData()}
+ */
+public class PersonBountyEventDataRepository {
+
+    private volatile List<String> factionsCOWlist;
+
+    private static volatile PersonBountyEventDataRepository instance;
+
+    // Private constructor to prevent instantiation
+    private PersonBountyEventDataRepository() {
+    }
+
+    // Double-checked locking for thread-safe singleton instance
+    public static PersonBountyEventDataRepository getInstance() {
+        if (instance == null) {
+            synchronized (PersonBountyEventDataRepository.class) {
+                if (instance == null) {
+                    instance = new PersonBountyEventDataRepository();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Synchronized wrapper around {@link SharedData#getData()} and subsequent {@link SharedData#getPersonBountyEventData()}
+     * @return the underlying {@link PersonBountyEventData}
+     */
+    public synchronized PersonBountyEventData getPersonBountyEventData() {
+        return SharedData.getData().getPersonBountyEventData();
+    }
+
+    /**
+     * Replacement for {@link SharedData#getPersonBountyEventData()} and subsequent {@link PersonBountyEventData#getParticipatingFactions()}
+     * underlying list
+     * <p>
+     * <b>NOTE: do not modify this list directly or through it's iterator, use the provided methods for that</b>
+     * @return unmodifiable CopyOnWriteArrayList wrapping the underlying list provided by the game
+     */
+    public synchronized List<String> getParticipatingFactions() {
+
+        if (factionsCOWlist == null) {
+            instantiateInternalCOWList();
+        }
+
+        if (!underlyingListHasSameContents()) {
+            instantiateInternalCOWList();
+        }
+
+        return factionsCOWlist;
+    }
+
+    /**
+     * Method that instantiates a new unmodifiable {@link CopyOnWriteArrayList} holding all participating bounty event factions.
+     * It's unmodifyable to make you not use it's iterator and remove stuff while iterating over it, but rather
+     * use a different removal technique through this <i>Repository</i>
+     */
+    private synchronized void instantiateInternalCOWList() {
+        factionsCOWlist = Collections.unmodifiableList(new CopyOnWriteArrayList<>(getPersonBountyEventData().getParticipatingFactions()));
+    }
```

Comment:
why... why is this an `unmodifiableList` wrapping a COWArrayList which is **made** to handle synchronization issues?

Reply:
well well well
```
    /**
     * Method that instantiates a new unmodifiable {@link CopyOnWriteArrayList} holding all participating bounty event factions.
     * It's unmodifyable to make you not use it's iterator and remove stuff while iterating over it, but rather
     * use a different removal technique through this <i>Repository</i>
     */
```

i guess i answered my question before i even asked it.

Reply:
resolving.